pkg/proc/variables.go

```diff
@@ -1341,9 +1374,27 @@ func readStringValue(mem MemoryReadWriter, addr uintptr, strlen int64, cfg LoadC
 		return "", fmt.Errorf("could not read string at %#v due to %s", addr, err)
 	}
 
-	retstr := *(*string)(unsafe.Pointer(&val))
+	//retstr := *(*string)(unsafe.Pointer(&val))
 
-	return retstr, nil
+	return string(val), nil
+}
+
+func readCStringValue(mem MemoryReadWriter, addr uintptr, cfg LoadConfig) (string, bool, error) {
+	done := false
+	buf := make([]byte, 1)
```

Comment:
Reading a single byte at a time until we hit a nul terminator is going to cause a lot of syscall context switching. Might be a premature optimization, but would be preferable to read into a larger buffer and search for the index of the nul terminator instead of going byte by byte.

Reply:
Done. It's a bit complicated because of pagefaults.